fulltext/backends/__gz.py

```diff
@@ -0,0 +1,32 @@
+import gzip
+from os.path import splitext, basename
+
+from fulltext import get, backend_from_fname, backend_from_fobj, is_file_path
+
+
+def _orig_fname(s):
+    s = basename(s)
+    if s.lower().endswith('.gz'):
+        s = s[:-3]
+    return s
+
+
+def _has_ext(s):
+    return bool(splitext(s)[1])
+
+
+def _get_file(path_or_file, **kwargs):
+    if is_file_path(path_or_file):
+        f = gzip.GzipFile(path_or_file)
+        path = path_or_file
+    else:
+        f = gzip.GzipFile(fileobj=path_or_file)
+        path = f.name
```

Comment:
If you define both backend functions (`_get_file()` and `_get_path()`) fulltext will figure out which to call, you can trust that you get a file-like object, or a path, depending on which function is called. You won't have to do any detection.